manifests/repo.pp

```diff
@@ -25,20 +29,17 @@
 #   }
 #
 define git::repo (
-  $target,
-  $bare    = false,
-  $source  = false,
-  $user    = 'root',
-  $group   = 'root',
-  $mode    = '0755',
-  $workdir = '/tmp',
-  $args    = undef,
-  $bin     = $git::params::bin,
+  String $target,
+  Boolean $bare = false,
+  Boolean $source = false,
+  String $user = 'root',
+  String $group = 'root',
+  String $mode = '0755',
+  String $workdir = '/tmp',
+  Optional[String] $args = undef,
```

Comment:
I know, but I didn't think it was worth the extra effort while this is good enough.